### PR TITLE
[flutter_tools] remove trailing eth info from fuchsia package server

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

This is not valid ipv6

### CODE REVIEW

File: packages/flutter_tools/lib/src/fuchsia/fuchsia_pm.dart [hunk 1]
```diff
@@ -180,7 +180,14 @@ class FuchsiaPM {
 ///   server.stop();
 /// }
 class FuchsiaPackageServer {
-  FuchsiaPackageServer(this._repo, this.name, this._host, this._port);
+  factory FuchsiaPackageServer(String repo, String name, String host, int port) {
+    if (host.contains('%')) {
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/fuchsia/fuchsia_pm.dart:184]: can you add a comment here?

**Review by christopherfujino — APPROVED** [review]
nit for a comment, otherwise LGTM

File: packages/flutter_tools/lib/src/fuchsia/fuchsia_pm.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/fuchsia/fuchsia_pm.dart:187]: Done

**Comment by jonahwilliams** [comment]
Landing on red to fix build, hopefully

**Comment by zanderso** [comment]
Can you explain this change a bit more? The trailing info is valid there IIRC. See in particular under "Use of zone indices in URIs" in https://en.wikipedia.org/wiki/IPv6_address.

**Comment by zanderso** [comment]
Also, can you link to the github issue that this is fixing? Thanks!

**Comment by jonahwilliams** [comment]
Maybe it was the URI formatting that was throwing it off?

specifically https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8882089532278067040/+/steps/Run_Fuchsia_Driver_Tests/0/steps/collect/0/logs/task_stdout_stderr:_flutter_fuchsia_driver_tests/0

```
[ +726 ms] executing: ssh -F /b/s/w/ir/fuchsia_ssh_config 192.168.42.104 amber_ctl add_repo_cfg -n flutter_tool -f http://[fe80::ec4:7aff:fecc:ea8f%25eno2]:34283/config.json
[  +48 ms] 2020/04/25 01:14:37 error adding repo: failed to GET file: Get "http://[fe80::ec4:7aff:fecc:ea8f%25eno2]:34283/config.json": dial tcp [fe80::ec4:7aff:fecc:ea8f%eno2]:34283: errno(111)
[        ] Failed to teach amber about the package server
```

**Comment by jonahwilliams** [comment]
ahh, `%25`, I missed that last time

**Pull request merged**